1624: Add role for verein360 to create api tokens

ztefanie opened the pull request:
### Short description
Added a new role for verein360 admins

### Proposed changes
- added a new role to create api tokens for verein360
- added new column "type" for api tokens
- adjusted project settings
- new external-user role can only create tokens of type "verified_application"
- project admins of koblenz create tokens of type "user_import"
- only api tokens of type user-import are allowed to use koblenz csv endpoint

### Side effects
- may break anything releated to api token generation of koblenz project admins or csv upload endpoint

### Testing
- create and delete tokens for different roles, check if type is set correctly
- check if endpoint for koblenz csv shows forbidden exception when used with api token of type external-verified
- add and remove user with external-api role

### Resolved issues

<!-- List all issues which should be closed when this PR is merged. -->

Fixes: #1624

f1sh1918 commented on administration/src/bp-modules/NavigationBar.tsx:
```diff
@@ -69,7 +69,7 @@ const Navigation = ({ onSignOut }: Props): ReactElement => {
           </NavLink>
         ) : null}
-        {role === Role.ProjectAdmin && config.userImportApiEnabled ? (
+        {(role === Role.ProjectAdmin && config.userImportApiEnabled) || role === Role.ExternalVerifiedApiUser ? (
```
:wrench:
I think it would improve the readability if we introduce a variable like
`canSeeProjectSettings` or sth similar

f1sh1918 reviewed (commented):
The concept looks really good to me 👍 
Just found some issues

<img width="398" alt="image" src="https://github.com/user-attachments/assets/8f9b9e4d-952f-4fcd-84bc-1ffc808e2048">

1. In the dashboard the item for "Projekt verwalten" is missing when i login with external api user

2. When i click on "Projekt verwalten" i get a confusing headline
<img width="530" alt="image" src="https://github.com/user-attachments/assets/9ca44a6b-0faf-4a18-a537-dd025bc5c50f">

I think the headline should be somehow connected with the user role to indicate the purpose of the token dynamically

3. Is it intended that Koblenz can also create an external api user?
I mean it could make sense to separate the user import right from project admin to a specific role (external api users) but is in our scenario for koblenz not really needed
<img width="451" alt="image" src="https://github.com/user-attachments/assets/76731126-8236-4367-8ddf-9f7d53c38135">

I think it could make sense to even a check in the backend that only certain ROLES and projects can `mayCreateExternalAPIUser` same for edit

ztefanie commented:
> 1. In the dashboard the item for "Projekt verwalten" is missing when i login with external api user

good find, fixed it.

> 2. When i click on "Projekt verwalten" i get a confusing headline
> 
fixed this by chaing the headling to "Api token"

> 
> 3. Is it intended that Koblenz can also create an external api user?
>    I mean it could make sense to separate the user import right from project admin to a specific role (external api users) but is in our scenario for koblenz not really needed
> 
> I think it would make sense to even add a check in the backend that only certain ROLES and projects can create external api user f.e. `mayCreateExternalAPIUser` fkt same for edit

Good point. I added this check to the existing "mayCreateUser" function.

f1sh1918 commented on administration/src/bp-modules/users/RoleSelector.tsx:
```diff
@@ -12,14 +13,21 @@ const RoleSelector = ({
+        Auswählen...
+      </option>
+      {hideProjectAdmin ? null : <option value={Role.ProjectAdmin}>{roleToText(Role.ProjectAdmin)}</option>}
+      {hideProjectAdmin || !config.applicationFeature?.applicationUsableWithApiToken ? null : (
+        <option value={Role.ExternalVerifiedApiUser}>{roleToText(Role.ExternalVerifiedApiUser)}</option>
```
```suggestion
     {config.applicationFeature?.applicationUsableWithApiToken && currentUserRole === Role.ProjectAdmin ? (
        <option value={Role.ExternalVerifiedApiUser}>{roleToText(Role.ExternalVerifiedApiUser)}</option>
      ) : null}
```
I think your check is really hard to read and even we need a line code more i think we should add the context for the currentUserRole and check that
```js
const { role: currentUserRole } = useContext(WhoAmIContext).me!
```

f1sh1918 reviewed (approved):
One small change i would really recommend because i found it difficult to understand the code.
Works fine. Nicely done 👍

seluianova commented on administration/src/bp-modules/project-settings/UserEndpointSettings.tsx:
```diff
@@ -49,7 +49,10 @@ const DeleteIcon = styled(Delete)`
 `
 
-const UserEndpointSettings = (): ReactElement => {
+type UserEndpointSettingsProps = {
```
:upside_down_face: I think it's not "user endoint settings" anymore
ProjectSettings / ApiTokenSettings ?

seluianova commented:
:upside_down_face: description of the new role is missing here
![image](https://github.com/user-attachments/assets/bbc437ae-8834-4a93-8040-5dc1db7a43ae)
:thought_balloon: also, maybe we want to disable this role for projects where it's not needed? (not necessarily as part of this PR but in general)

seluianova reviewed (approved):
lgtm, just shared a couple of minor thoughts
